### DISCUSSION TITLE
Would love an example of logging into AAD B2C

### DISCUSSION THREAD

**Discussion by blackgrouse** [cypress-io/cypress-realworld-app#872] [Ideas]
Our app uses Azure Active Directory B2C for user auth and I'm currently logging in using a `cy.task` which uses Puppeteer to login for us outside of Cypress.

Would love to be able to do this via `cy.response`!  Do you guys have any plans for an example of how to do this?

**Comment by kevinold** [comment]
@blackgrouse Yes, we have begun to experiment with Active Directory in the RWA.

A [partial implementation branch is here](https://github.com/cypress-io/cypress-realworld-app/compare/kevin/microsoft-auth). The core is this [Cypress command](https://github.com/cypress-io/cypress-realworld-app/compare/kevin/microsoft-auth#diff-6c107e2390c5ad84e123f3be09d7b91ba35de8f8cc550d1f358c75806df47645R323)

Note, you must [create an Azure AD account](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/add-users-azure-active-directory) for the testing user intended to be use with this command.

At the moment, the Microsoft React components are in alpha and we cannot rely on it's API to complete the signin process.

**Reply by blackgrouse** [reply]
@kevinold Awesome news, I'll keep an eye out for it, thanks!

**Reply by kevinold** [reply]
@blackgrouse That said, most of the functionality is present in the branch.  Does the command I pointed to on the branch help?

**Reply by blackgrouse** [reply]
@kevinold Unfortunately, AAD B2C has a slightly different flow which I haven't got working yet in Cypress.  [More info](https://docs.microsoft.com/en-us/azure/active-directory-b2c/add-ropc-policy?tabs=app-reg-ga&pivots=b2c-user-flow)

**Comment by kevinold** [comment]
@blackgrouse Please refer to our [Auth0 custom command](https://github.com/cypress-io/cypress-realworld-app/blob/develop/cypress/support/auth-provider-commands/auth0.ts#L21-L53) which implements the resource owner password credentials (ROPC) flow for the AAD B2C you pointed to.  Does this help?
